### PR TITLE
[CLK32-957]: Changepass API + Logic

### PR DESCRIPTION
**Pull request by Minhdn1**
## Prerequisite
- [x] Self-review :monocle_face:
- [x] No Crashes ☠️
---

### Root cause
- // Explain what you have done

### Solution
- // Explain how you fix it

### Impact
- // List of affected components/modules

### Screenshots (Optional)
- // Capture affected screens

### CODE REVIEW

File: ClearKeep/Scenes/ChangePassword/Views/ChangePasswordContentView.swift [hunk 1]
```diff
@@ -80,104 +106,105 @@ private extension ChangePasswordContentView {
...
 			Task {
-				await injected.interactors.changePasswordInteractor.resetPassword(preAccessToken: preAccessToken, email: email, rawNewPassword: newPassword, domain: domain)
+				do {
+					let result = try await injected.interactors.changePasswordInteractor.changePassword(oldPassword: currentPassword, newPassword: newPassword).get().authenResponse?.error
+					self.messageAlert = result ?? "NewPassword.Sucess".localized
+					self.isShowAlert = true
+				} catch {
+					self.messageAlert = "\(error)"
+					self.isShowAlert = true
+//					self.messageAlert = "NewPassword.Sucess".localized
```
Comment (vmo-klinh) [inline ClearKeep/Scenes/ChangePassword/Views/ChangePasswordContentView.swift]: remove unused code

File: ClearKeep/Scenes/ChangePassword/Views/ChangePasswordContentView.swift [hunk 2]
```diff
@@ -80,104 +106,105 @@ private extension ChangePasswordContentView {
...
 		checkInvalid = injected.interactors.changePasswordInteractor.checkValid(passwordValdid: passwordInvalid, confirmPasswordValid: confirmPasswordInvvalid)
 	}
+
+	func disableButton() -> Bool {
+		if checkInvalid || currentInvalid {
+			return false
+		} else {
+			return true
+		}
+	}
```
Comment (vmo-klinh) [inline ClearKeep/Scenes/ChangePassword/Views/ChangePasswordContentView.swift:145]: return !(checkInvalid || currentInvalid)

File: ClearKeep/App/Resources/Localize.strings [hunk 3]
```diff
@@ -100,7 +102,12 @@
 // MARK: - NewPassword
 "NewPassword.Title" = "Enter Your New Password";
 "NewPassword.Description" = "Please enter your details to change password";
-"NewPassword.NewPassword" = "New Password";
+"NewPassword.New.PlaceHold" = "New Password";
+"NewPassword.Curren.PlaceHold" = "Current Password";
```
Comment (vmo-klinh) [inline ClearKeep/App/Resources/Localize.strings]: typo 'current'

File: ClearKeep/Scenes/ChangePassword/Views/ChangePasswordContentView.swift (hunk 1 above)
Comment (Minhdn1) [reply ClearKeep/Scenes/ChangePassword/Views/ChangePasswordContentView.swift]: fix complete

File: ClearKeep/App/Resources/Localize.strings (hunk 3 above)
Comment (Minhdn1) [reply ClearKeep/App/Resources/Localize.strings]: fix complete

File: ClearKeep/Scenes/ChangePassword/Views/ChangePasswordContentView.swift (hunk 2 above)
Comment (Minhdn1) [reply ClearKeep/Scenes/ChangePassword/Views/ChangePasswordContentView.swift:145]: fix complete

**Review by vmo-klinh — APPROVED** [review]

**Pull request merged**